Sources_v3/StreamChatUI/Utils/UITextView+Extensions.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+extension UITextView {
+    ///
+    /// Reference:
+    /// https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TextLayout/Tasks/StringHeight.html
+    ///
+    var calculatedTextHeight: CGFloat {
+        let textStorage = NSTextStorage(string: text)
+        let width = frame.width - textContainerInset.right - textContainerInset.left
+        let customTextContainer = NSTextContainer(size: .init(width: width, height: CGFloat.greatestFiniteMagnitude))
+        let layoutManager = NSLayoutManager()
```

Comment:
Why do you create new instances of `NSTextContainer` and `NSLayoutManager`? Can't we just use to _actual_ container and manager of the text view?

Reply:
Cause we are creating artificial `NSTextContainer` here with our custom size.
We don't want to change anything about current `UITextView`.